src/components/CouponDetails/helpers.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable import/prefer-default-export */
+import { SINGLE_USE, ONCE_PER_CUSTOMER } from '../../data/constants/coupons';
+import {
+  ACTION_LABELS, ACTION_TYPES, COUPON_FILTER_TYPES, FILTER_OPTIONS,
+} from './constants';
+
+export const getFilterOptions = (usageLimitation) => {
+  const shouldHidePartialRedeem = [SINGLE_USE, ONCE_PER_CUSTOMER].includes(usageLimitation);
+  let options = FILTER_OPTIONS;
+
+  if (shouldHidePartialRedeem) {
+    options = FILTER_OPTIONS.filter(option => option.value !== COUPON_FILTER_TYPES.partiallyRedeemed);
+  }
+
+  return options;
+};
+
+export const getFirstNonDisabledOption = (options) => {
+  const firstNonDisabledOption = options.find(option => !option.disabled);
```

Comment:
Do we care about order of operations here? Ie will the "first" option always be the same and does that matter?

Reply:
Yeah, the first option will always be the same, we never re-order the options.

Reply:
Interesting. I figured Javascript might just arbitrarily pull options rather than treating the options like an index in a list (ie immovable)